spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -24,62 +24,74 @@
 import com.google.api.gax.grpc.InstantiatingExecutorProvider;
 import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.pubsub.spi.v1.Publisher;
-import com.google.protobuf.ByteString;
 import com.google.pubsub.v1.PubsubMessage;
 import com.google.pubsub.v1.TopicName;
 
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.cloud.gcp.pubsub.integration.converters.SimpleMessageConverter;
 import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.MessageConversionException;
 import org.springframework.messaging.converter.MessageConverter;
 
 /**
  * @author Vinicius Carvalho
  */
 public class PubSubTemplate implements PubSubOperations, InitializingBean {
 
-	private final String projectId;
+  private final String projectId;
 
-	private final GoogleCredentials credentials;
+  private final GoogleCredentials credentials;
 
-	private ConcurrentHashMap<String, Publisher> publishers = new ConcurrentHashMap<>();
+  private ConcurrentHashMap<String, Publisher> publishers = new ConcurrentHashMap<>();
 
-	private ExecutorProvider executorProvider;
+  private ExecutorProvider executorProvider;
 
-	private MessageConverter converter;
+  private MessageConverter messageConverter;
 
-	private int concurrentProducers = 1;
+  private int concurrentProducers = 1;
 
-	public PubSubTemplate(GoogleCredentials credentials, String projectId){
-		this.projectId = projectId;
-		this.credentials = credentials;
-		this.executorProvider = InstantiatingExecutorProvider.newBuilder().setExecutorThreadCount(concurrentProducers).build();
-		this.converter = new SimpleMessageConverter();
-	}
+  public PubSubTemplate(GoogleCredentials credentials, String projectId) {
+    this(credentials, projectId, new SimpleMessageConverter());
+  }
 
-	@Override
-	public String send(final String topic, Message message) throws RuntimeException{
+  public PubSubTemplate(GoogleCredentials credentials, String projectId,
+      MessageConverter messageConverter) {
```

Comment:
It is optional, you can use the first constructor if you don't want to specify a converter.
I didn't specify a setter so that the object is immutable.